src/csharp/Grpc.Examples/MathGrpc.cs

```diff
@@ -27,10 +27,73 @@ public static partial class Math
   {
     static readonly string __ServiceName = "math.Math";
 
+    #if !GOOGLE_PROTOBUF_DISABLE_BUFFER_SERIALIZATION
+    static readonly grpc::Marshaller<global::Math.DivArgs> __Marshaller_math_DivArgs = 
+      grpc::Marshallers.Create(
+        (arg, context) =>
+        {
+          var writer = new global::Google.Protobuf.CodedOutputWriter(context.GetBufferWriter());
+          arg.WriteTo(ref writer);
```

Comment:
How much should we accommodate the scenario where the messages are built separate from the services and isn't an `IBufferMessage`? This line won't compile and the project needs to define `GOOGLE_PROTOBUF_DISABLE_BUFFER_SERIALIZATION`

Reply:
GOOGLE_PROTOBUF_DISABLE_BUFFER_SERIALIZATION is one option. Another is to turn off buffer serialization either for this proto file or for all protofiles.

I don't think a runtime check is needed. Also we have the ability to change codegen to add it later if it turns out that it is a common problem.

Reply:
Ok I'll remove the `GOOGLE_PROTOBUF_DISABLE_BUFFER_SERIALIZATION` check. Agreed that runtime check isn't necessary.